gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -161,6 +167,35 @@ public void contentChanged() {
         };
     }
 
+    private void handleEvent(NodeEvent nodeEvent) {
+        ProjectFile projectFile = project.getFileSystem().findProjectFile(nodeEvent.getId(), ProjectFile.class);
+        if (NodeDataUpdated.TYPENAME.equals(nodeEvent.getType()) || DependencyAdded.TYPENAME.equals(nodeEvent.getType())) {
+            refreshProjectFile(projectFile);
+        } else if (NodeNameUpdated.TYPENAME.equals(nodeEvent.getType())) {
+            List<ProjectFile> backwardDependencies = projectFile.getBackwardDependencies();
+            backwardDependencies.forEach(this::refreshProjectFile);
+        }
+    }
+
+    private void refreshProjectFile(ProjectFile projectFile) {
+        if (projectFile != null) {
+            List<TreeItem<Object>> allVisibleTreeItems = new ArrayList<>();
+            findAllVisibleTreeItems(treeView.getRoot(), allVisibleTreeItems);
+            Optional<TreeItem<Object>> projectFileItem = allVisibleTreeItems.stream()
+                    .filter(item -> ((ProjectNode) item.getValue()).getId().equals(projectFile.getId()))
+                    .findFirst();
+            projectFileItem.ifPresent(treeItem -> refresh(treeItem.getParent()));
+        }
+    }
+
+    private void findAllVisibleTreeItems(TreeItem<Object> folderTreeItem, List<TreeItem<Object>> allVisibleTreeItems) {
+        ObservableList<TreeItem<Object>> children = folderTreeItem.getChildren();
+        allVisibleTreeItems.addAll(children);
+        children.stream()
+                .filter(item -> ((ProjectNode) item.getValue()).isFolder() && item.isExpanded())
```

Comment:
must be done on all nodes even those who ain't visible

Reply:
resolved